test/worker_custom_test.go

```diff
@@ -71,3 +78,101 @@ func TestCustomWorker(t *testing.T) {
 	time.Sleep(10 * time.Second)
 	worker.Shutdown()
 }
+
+func TestWorkerInjectKinesis(t *testing.T) {
+	kclConfig := cfg.NewKinesisClientLibConfig("appName", streamName, regionName, workerID).
+		WithInitialPositionInStream(cfg.LATEST).
+		WithMaxRecords(10).
+		WithMaxLeasesForWorker(1).
+		WithShardSyncIntervalMillis(5000).
+		WithFailoverTimeMillis(300000).
+		WithMetricsBufferTimeMillis(10000).
+		WithMetricsMaxQueueSize(20)
+
+	log.SetOutput(os.Stdout)
+	log.SetLevel(log.DebugLevel)
+
+	assert.Equal(t, regionName, kclConfig.RegionName)
+	assert.Equal(t, streamName, kclConfig.StreamName)
+
+	// configure cloudwatch as metrics system
+	metricsConfig := getMetricsConfig(kclConfig, metricsSystem)
+
+	// create custom Kinesis
+	s, err := session.NewSession(&aws.Config{
+		Region: aws.String(regionName),
+	})
+	assert.Nil(t, err)
+	kc := kinesis.New(s)
+
+	// Inject a custom checkpointer into the worker.
+	worker := wk.NewWorker(recordProcessorFactory(t), kclConfig, metricsConfig).
+		WithKinesis(kc)
+
+	err = worker.Start()
+	assert.Nil(t, err)
+
+	// Put some data into stream.
+	for i := 0; i < 100; i++ {
+		// Use random string as partition key to ensure even distribution across shards
+		err := worker.Publish(streamName, utils.RandStringBytesMaskImpr(10), []byte(specstr))
+		if err != nil {
+			t.Errorf("Errorin Publish. %+v", err)
+		}
+	}
+
+	// wait a few seconds before shutdown processing
+	time.Sleep(10 * time.Second)
+	worker.Shutdown()
+}
+
+func TestWorkerInjectKinesisAncCheckpointer(t *testing.T) {
```

Comment:
`TestWorkerInjectKinesisAncCheckpointer` should be `TestWorkerInjectKinesisAndCheckpointer`

Reply:
Fixed. thx.